assets/js/modules/sign-in-with-google/components/common/ClientIDTextField.js

```diff
@@ -53,27 +58,65 @@ export default function ClientIDTextField() {
 
 			if ( newValue !== clientID ) {
 				setClientID( newValue );
+				setIsExistingClientID( false );
 			}
 
 			debounceSetIsValid( isValidClientID( newValue ) );
 		},
 		[ clientID, setClientID, debounceSetIsValid ]
 	);
 
+	// Prefill the clientID field with a value from a previous module connection, if it exists.
+	useMount( async () => {
+		// Allow default `settings` and `savedSettings` to load before updating
+		// the `clientID` setting again.
+		await registry
+			.resolveSelect( MODULES_SIGN_IN_WITH_GOOGLE )
+			.getSettings();
+
+		// The clientID is fetched again as useMount does not receive the
+		// updated clientID.
+		const currentClientID = registry
+			.select( MODULES_SIGN_IN_WITH_GOOGLE )
+			.getClientID();
+
+		if (
+			currentClientID === '' &&
+			global._googlesitekitModulesData?.[ 'sign-in-with-google' ][
+				'existingClientID'
+			]
+		) {
+			setClientID(
+				global._googlesitekitModulesData[ 'sign-in-with-google' ]
+					.existingClientID
+			);
+			setIsExistingClientID( true );
+		}
+	} );
```

Comment:
We need to move it to the setup form component and pass the existing client id as a property to this component.

Reply:
Apologies again, but I don't quite understand. The existing client ID is a global anyway, right? So what is the point of passing it as a prop? We don't need any of the information computed here within SetupForm, so is there a reason to move this one level up?

Reply:
Thanks - of course, ClientIDTextField is being used in other forms!